lib/atomic/activities.ex

```diff
@@ -233,6 +233,27 @@ defmodule Atomic.Activities do
   """
   def get_enrollment!(id), do: Repo.get!(Enrollment, id)
 
+  def get_enrollment(activity, user) do
+    Enrollment
+    |> where(activity_id: ^activity.id, user_id: ^user.id)
+    |> Repo.one()
+  end
+
+  def get_user_enrolled(user, activity) do
+    enrollment =
+      Enrollment
+      |> where(user_id: ^user.id, activity_id: ^activity.id)
+      |> Repo.one()
+
+    if enrollment do
+      Repo.delete(enrollment)
+      {:ok, "Participation canceled"}
+    else
+      create_enrollment(activity, user)
+      {:ok, "Participation confirmed"}
+    end
+  end
```

Comment:
```suggestion
  def switch_user_enrolled(user, activity) do
    enrollment =
      Enrollment
      |> where(user_id: ^user.id, activity_id: ^activity.id)
      |> Repo.one()

    if enrollment do
      Repo.delete(enrollment)
      :cancelled
    else
      create_enrollment(activity, user)
      :enrolled
    end
  end
```

The name as it is is pretty confusing (I am not particularly happy with my suggestion either). Regardless, we should document this function to make sure its functionality is understood.

Also, if you are always returning `{:ok, something}`, might as well just return the `something`, as this will never fail

Reply:
You're right, thanks !!